debian/srptools.init

```diff
@@ -9,83 +9,126 @@
 # Description:       Discovers SRP scsi over infiniband targets.
 ### END INIT INFO
 
-[ -x /usr/sbin/srp_daemon ] || exit 0
-
+DAEMON=/usr/sbin/srp_daemon
 IBDIR=/sys/class/infiniband
 
 PORTS=""
 RETRIES=""
+RETRIES_DEFAULT=60
 LOG=""
+LOG_DEFAULT=/var/log/srp_daemon.log
+
+[ -x $DAEMON ] || exit 0
 
 . /lib/lsb/init-functions
 
 [ -f /etc/default/srptools ] &&  . /etc/default/srptools
 
-start_daemon () {
+max() {
+    echo $(($1 > $2 ? $1 : $2))
+}
 
-if [ "$PORTS" = "NONE" ] ; then
-echo "srptools disabled."
-exit 0
-fi
+run_daemon() {
+    local STATUS
 
+    # srp_daemon wedges if we start it on a port which is not up
 
-if [ "$PORTS" = "ALL" ]  ; then
-    for HCA_ID in `/bin/ls -1 ${IBDIR}`
-      do
-      for PORT in `/bin/ls -1 ${IBDIR}/${HCA_ID}/ports/`
-        do
-        run_daemon
-      done
-    done
-fi
+    STATUS=$(<"${IBDIR}/${HCA_ID}/ports/${PORT}/state")
```

Comment:
Failing to start the daemon due to port state is racey and will lead to boot up failures. Ultimately srp_daemon itself has to monitor the port state...